File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res=[]
        # for i in range (len(nums)):
        #     for j in range(i+1,len(nums)):
        #         if nums[i]+nums[j]==target:
        #             res.append(i)
        #             res.append(j)
        #             return res
        # return res
        m={}#dictionary to store index
        res=[]
        for i in range(len(nums)):
            diff=target-nums[i]
            if diff in m:
                res=[m[diff],i]
            m[nums[i]]=i
        return res

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> ht=new HashMap<>();
        int[] res=new int[2];
        int comp;
        for(int i=0;i<nums.length;i++){
            comp=target-nums[i];
            if( ht.containsKey(comp)){
                res[0]=ht.get(comp);
                res[1]=i;
            }
            ht.put(nums[i],i);
        }
        return res;
    }
}